### PR TITLE
Add 4.14 eus upgrade on amd64/arm64

### PR DESCRIPTION
**Pull request by shellyyang1989**
@jianlinliu @jiajliu @JianLi-RH @jhou1 for review. Thank you!

### CODE REVIEW

File: ci-operator/jobs/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14-periodics.yaml [hunk 1]
```diff
@@ -20132,9 +20133,9 @@ periodics:
       - --oauth-token-path=/usr/local/github-credentials/oauth
       - --report-credentials-file=/etc/report/credentials
       - --secret-dir=/secrets/ci-pull-credentials
-      - --secret-dir=/usr/local/aws-ipi-p1-f14-cluster-profile
-      - --target=aws-ipi-p1-f14
-      - --variant=amd64-nightly-4.14-upgrade-from-stable-4.11
+      - --secret-dir=/usr/local/nutanix-ipi-proxy-fips-p2-f28-cluster-profile
+      - --target=nutanix-ipi-proxy-fips-p2-f28
```
Comment (jiajliu) [inline ci-operator/jobs/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14-periodics.yaml:20137]: There are known [issue](https://issues.redhat.com/browse/OCPQE-15938) for 4.12-4.13 upgrade on Nutanix.  Maybe we could delay the eus test on such platform.

Comment (shellyyang1989) [reply ci-operator/jobs/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14-periodics.yaml:20137]: Good suggestion! Lower the test frequency to f360 and let's move it back after nutanix upgrade step is ready.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__multi-nightly-4.14-upgrade-from-eus-4.12.yaml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        and @network-ovnkubernetes and not @fips'
+    test:
+    - ref: ipi-install-heterogeneous
+    - chain: openshift-upgrade-qe-test-paused
+    workflow: cucushift-installer-rehearse-aws-ipi-ovn-ipsec
+zz_generated_metadata:
+  branch: release-4.14
+  org: openshift
+  repo: openshift-tests-private
+  variant: multi-nightly-4.14-upgrade-from-eus-4.12
```
[the remark this answers is not in the document]
Comment (shellyyang1989) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__multi-nightly-4.14-upgrade-from-eus-4.12.yaml]: Right, it's tp in 4.12. Removed.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__arm64-nightly-4.14-upgrade-from-eus-4.12.yaml [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+      OCP_ARCH: arm64
+      UPGRADE_CHECK_RUN_TAGS: '@arm64 and @upgrade-check and @4.14 and @aws-ipi and
+        not @fips'
+      UPGRADE_PRE_RUN_TAGS: '@arm64 and @upgrade-prepare and @4.14 and @aws-ipi and
+        not @fips'
+    test:
+    - ref: cucushift-installer-check
+    - chain: openshift-upgrade-qe-test-paused
+    workflow: cucushift-installer-rehearse-aws-ipi-ovn
+- as: gcp-ipi-workload-identity-rt-fips-f14
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__arm64-nightly-4.14-upgrade-from-eus-4.12.yaml:105]: On 4.12, only aws and azure support arm64 arch.

Comment (shellyyang1989) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__arm64-nightly-4.14-upgrade-from-eus-4.12.yaml]: Thanks for the comments! Updated.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__arm64-nightly-4.14-upgrade-from-eus-4.12.yaml [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+      COMPUTE_NODE_TYPE: m6g.xlarge
+      CONTROL_PLANE_INSTANCE_TYPE: m6g.xlarge
+      E2E_RUN_INTERMEDIATE_TAGS: '@arm64 and @upgrade-sanity and @4.13 and @aws-ipi
+        and not @fips'
+      E2E_RUN_LATEST_TAGS: '@arm64 and @upgrade-sanity and @4.14 and @aws-ipi and
+        not @fips'
+      OCP_ARCH: arm64
+      UPGRADE_CHECK_RUN_TAGS: '@arm64 and @upgrade-check and @4.14 and @aws-ipi and
+        not @fips'
+      UPGRADE_PRE_RUN_TAGS: '@arm64 and @upgrade-prepare and @4.14 and @aws-ipi and
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__arm64-nightly-4.14-upgrade-from-eus-4.12.yaml:99]: should be "@4.12" here?

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly-4.14-upgrade-from-eus-4.12.yaml [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+        NETWORK_TYPE=OVNKubernetes
+      E2E_RUN_INTERMEDIATE_TAGS: '@4.13 and @baremetal-ipi and @upgrade-sanity'
+      E2E_RUN_LATEST_TAGS: '@4.14 and @baremetal-ipi and @upgrade-sanity'
+      UPGRADE_CHECK_RUN_TAGS: '@upgrade-check and @4.14 and @baremetal-ipi'
+      UPGRADE_PRE_RUN_TAGS: '@upgrade-prepare and @4.14 and @baremetal-ipi'
+    test:
+    - ref: cucushift-installer-check
+    - chain: openshift-upgrade-qe-test-paused
+    workflow: cucushift-installer-rehearse-baremetalds-ipi-ovn-ipv4
+- as: gcp-ipi-workload-identity-rt-fips-f14
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly-4.14-upgrade-from-eus-4.12.yaml:80]: how about f28 to keep consistent with other jobs.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly-4.14-upgrade-from-eus-4.12.yaml [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+      product: ocp
+      stream: nightly
+      version: "4.14"
+resources:
+  '*':
+    requests:
+      cpu: 100m
+      memory: 200Mi
+tests:
+- as: baremetalds-ipi-ovn-ipv4-fips-p2-f28
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly-4.14-upgrade-from-eus-4.12.yaml:63]: Suggest to remove `priority` field in the PR, we are going to remove the field soon.

Comment (shellyyang1989) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly-4.14-upgrade-from-eus-4.12.yaml]: Good to know. Removed.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly-4.14-upgrade-from-eus-4.12.yaml (hunk 5 above)
Comment (shellyyang1989) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly-4.14-upgrade-from-eus-4.12.yaml]: Sure, updated.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__arm64-nightly-4.14-upgrade-from-eus-4.12.yaml (hunk 4 above)
Comment (shellyyang1989) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__arm64-nightly-4.14-upgrade-from-eus-4.12.yaml:99]: yeah, i think so. updated.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly-4.14-upgrade-from-eus-4.12.yaml [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+      E2E_RUN_INTERMEDIATE_TAGS: '@4.13 and @nutanix-ipi and @upgrade-sanity'
+      E2E_RUN_LATEST_TAGS: '@4.14 and @nutanix-ipi and @upgrade-sanity'
+      FIPS_ENABLED: "true"
+      UPGRADE_CHECK_RUN_TAGS: '@upgrade-check and @4.14 and @nutanix-ipi'
+      UPGRADE_PRE_RUN_TAGS: '@upgrade-prepare and @4.12 and @nutanix-ipi'
+    test:
+    - ref: cucushift-installer-check
+    - chain: openshift-upgrade-qe-test-paused
+    workflow: cucushift-installer-rehearse-nutanix-ipi-proxy
+- as: vsphere-ipi-f14
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly-4.14-upgrade-from-eus-4.12.yaml:126]: how about f28 too here.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__arm64-nightly-4.14-upgrade-from-eus-4.12.yaml [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+      RELEASE_IMAGE_TARGET: release:target-arm64
+    env:
+      BASE_DOMAIN: qe.azure.devcluster.openshift.com
+      COMPUTE_NODE_TYPE: Standard_D4ps_v5
+      E2E_RUN_INTERMEDIATE_TAGS: '@arm64 and @4.13 and @azure-ipi and @upgrade-sanity
+        and not @fips'
+      E2E_RUN_LATEST_TAGS: '@arm64 and @4.14 and @azure-ipi and @upgrade-sanity and
+        not @fips'
+      OCP_ARCH: arm64
+      UPGRADE_CHECK_RUN_TAGS: '@arm64 and @4.14 and @azure-ipi and @upgrade-check
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__arm64-nightly-4.14-upgrade-from-eus-4.12.yaml:121]: sounds `@4.12` here?

Comment (shellyyang1989) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__arm64-nightly-4.14-upgrade-from-eus-4.12.yaml:121]: ah, makes sense. it should check 4.12 features working as expected. Thanks for the careful review

**Comment by shellyyang1989** [comment]
@jiajliu @jianlinliu needs an extra lgtm as it's rebased:)

**Pull request merged**